migrations/2025-02-05-083109_add-num-versions-column/up.sql

```diff
@@ -0,0 +1,27 @@
+ALTER TABLE default_versions
+    ADD COLUMN num_versions INTEGER;
+
+COMMENT ON COLUMN default_versions.num_versions IS 'The total number of versions.';
+
+CREATE OR REPLACE FUNCTION update_num_versions_from_versions() RETURNS TRIGGER AS $$
+BEGIN
+    IF (TG_OP = 'INSERT') THEN
+        INSERT INTO default_versions (crate_id, version_id, num_versions)
+        VALUES (NEW.crate_id, NEW.id, 1)
+        ON CONFLICT (crate_id) DO UPDATE
+        SET num_versions = EXCLUDED.num_versions + 1;
```

Comment:
is `EXCLUDED.num_versions` actually correct here? shouldn't this be `default_versions.num_versions`? 🤔 

maybe I'm missing something, but isn't `EXCLUDED.num_versions` referring to the `1` two lines above?

Reply:
Ah, 🤦 🙈 , it should definitely be `default_versions.num_versions + 1` Do you want me to send a patch migration, or will you take care of it?

Reply:
I'm currently busy with some other stuff, so if you have time, that would be great :)

Ideally with a regression test case 😅